x/gov/keeper/msg_server.go

```diff
@@ -31,6 +32,17 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *types.MsgSubmitPro
 		return nil, err
 	}
 
+	bytes, err := proposal.Marshal()
+	if err != nil {
+		return nil, err
+	}
+
+	// ref: https://github.com/cosmos/cosmos-sdk/issues/9683
+	ctx.BlockGasMeter().ConsumeGas(
```

Comment:
```suggestion
	ctx.GasMeter().ConsumeGas(
```

We shouldn't be using the BlockGasMeter here (resets on block), but the GasMeter (resets on each tx).

At the end of each tx, baseapp adds the GasMeter's `GasConsumed` to the block's BlockGasMeter, and resets GasMeter for the next tx